giles-eco-tardis/src/main/java/edu/asu/diging/tardis/core/service/impl/ImageExtractionManager.java

```diff
@@ -73,110 +90,86 @@ public void init() {
+            return;
         }
-
-        completedRequest.setDocumentId(request.getDocumentId());
-        completedRequest.setFileId(request.getFileId());
-        completedRequest.setNotifier(propertiesManager.getProperty(Properties.NOTIFIER_ID));
-        completedRequest.setStatus(status);
-        completedRequest.setExtractionDate(OffsetDateTime.now(ZoneId.of("UTC")).toString());
-        completedRequest.setPages(pages);
-
-        progressManager.setPhase(ProgressPhase.DONE);
         try {
-            requestProducer.sendRequest(completedRequest,
-                    propertiesManager.getProperty(Properties.KAFKA_TOPIC_COMPLETION_NOTIFICATIION));
-        } catch (MessageCreationException e) {
-            messageHandler.handleMessage("Could not send message.", e, MessageType.ERROR);
+            innogenScriptRunner.runInnogenScript(imagePath,request.getPageNr());
+        } catch (InnogenScriptRunnerException e) {
+            messageHandler.handleMessage("Could not run docker command" , e, MessageType.ERROR);
         }
-        
-        progressManager.reset();
+        Path path = Paths.get(imagePath);
+        String outputParentFolderPath = path.getParent().toString() + File.separator + propertiesManager.getProperty(Properties.EXTRACTED_FOLDER) + File.separator + request.getPageNr() + File.separator + propertiesManager.getProperty(Properties.EXTRACTED_FOLDER);
+        File outputDirectory = new File(outputParentFolderPath);
+        File[] files = outputDirectory.listFiles();
+        String restEndpoint = getRestEndpoint();
+        List<edu.asu.diging.gilesecosystem.requests.impl.Page> pages = new ArrayList<>();
+        edu.asu.diging.gilesecosystem.requests.impl.Page requestPage = new edu.asu.diging.gilesecosystem.requests.impl.Page();
+        requestPage.setPageElements(new ArrayList<>());
+        requestPage.setPageNr(request.getPageNr());
+        if (files != null && files.length > 0) {
+            for (File file : files) {
+                PageElement pageElem = new PageElement();
+                pageElem.setContentType("image/png");
+                pageElem.setFilename(file.getName());
+                pageElem.setType("IMAGE");
+                pageElem.setDownloadUrl(
+                restEndpoint + DownloadFileController.GET_FILE_URL
+                    .replace(DownloadFileController.USER_NAME_PLACEHOLDER, request.getUsername())
+                    .replace(DownloadFileController.UPLOAD_ID_PLACEHOLDER, request.getUploadId())
+                    .replace(DownloadFileController.DOCUMENT_ID_PLACEHOLDER, request.getDocumentId())
+                    .replace(DownloadFileController.PAGE_NR, String.valueOf(request.getPageNr()))
+                    .replace(DownloadFileController.FILENAME_PLACEHOLDER, file.getName()));
+                pageElem.setStatus(PageStatus.COMPLETE);
+                requestPage.getPageElements().add(pageElem);
```

Comment:
I would create a method for the code in the for loop and call that method in the loop